Add ability to check files one by one and print which are currently checked/parsed

When trying to found file which crashes, I use binary search so I:
1. run ruff to check for crash
2. remove half of files
3. run again to see if crash still happens
4. if yes, back to point 2
5. if no, restore half of files, remove other part, back to point 3

With such output I will be able to find crash with seconds instead minutes(of course this should work only on 1 thread):
```
[[ Parsing a.py ]]
[[ Parsing b.py ]]
[[ Parsing c.py ]]
[[ Parsing d.py ]]

[[ Checking a.py file ]]
a.py:21:9: D400 [*] First line should end with a period
[[ Checking a.py file ]]
b.py:21:9: D415 [*] First line should end with a period, question mark, or exclamation point
[[ Checking c.py file ]]
[[ Checking d.py file ]]
```

--- a/crates/ruff/src/resolver.rs
+++ b/crates/ruff/src/resolver.rs
@@ -128,7 +128,7 @@ pub fn resolve_configuration(
 
         // Resolve the current path.
         let options = pyproject::load_options(&path)
-            .map_err(|err| anyhow!("Failed to parse `{}`: {}", path.to_string_lossy(), err))?;
+            .map_err(|err| anyhow!("Failed to parse `{}`: {}", path.display(), err))?;
         let project_root = relativity.resolve(&path);
         let configuration = Configuration::from_options(options, &project_root)?;
 

--- a/crates/ruff_cli/src/commands/add_noqa.rs
+++ b/crates/ruff_cli/src/commands/add_noqa.rs
@@ -54,7 +54,7 @@ pub fn add_noqa(
             match add_noqa_to_path(path, package, settings) {
                 Ok(count) => Some(count),
                 Err(e) => {
-                    error!("Failed to add noqa to {}: {e}", path.to_string_lossy());
+                    error!("Failed to add noqa to {}: {e}", path.display());
                     None
                 }
             }

--- a/crates/ruff_cli/src/commands/run.rs
+++ b/crates/ruff_cli/src/commands/run.rs
@@ -45,10 +45,7 @@ pub fn run(
     if cache.into() {
         fn init_cache(path: &std::path::Path) {
             if let Err(e) = cache::init(path) {
-                error!(
-                    "Failed to initialize cache at {}: {e:?}",
-                    path.to_string_lossy()
-                );
+                error!("Failed to initialize cache at {}: {e:?}", path.display());
             }
         }
 

--- a/crates/ruff_cli/src/diagnostics.rs
+++ b/crates/ruff_cli/src/diagnostics.rs
@@ -74,13 +74,15 @@ pub fn lint_path(
         if let Some(messages) =
             cache::get(path, package.as_ref(), &metadata, settings, autofix.into())
         {
-            debug!("Cache hit for: {}", path.to_string_lossy());
+            debug!("Cache hit for: {}", path.display());
             return Ok(Diagnostics::new(messages));
         }
         Some(metadata)
     } else {
         None
     };
+
+    debug!("Checking: {}", path.display());
 
     // Read the file from disk.
     let contents = std::fs::read_to_string(path)?;

